lib/serverless-tools/deployer/ecr_pusher.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+require "aws-sdk-ecr"
+
+module ServerlessTools
+  module Deployer
+    class EcrPusher
+      def initialize(client:, git:, config:)
+        @client = client
+        @git = git
+        @config = config
+      end
+
+      def push(local_image_name:)
+        if image_tags.include?(tag)
```

Comment:
Just for visibility - I'd planned on moving this checks out into the layer above (the function deployer class) - it doesn't impact this PR, but when I get back to it - it should simplify the pusher classes a little bit.